test-packages/e2e-tests/test/generator-cli.spec.ts

```diff
@@ -30,53 +41,119 @@ describe('generator-cli', () => {
-      path.resolve(__dirname, 'generator.config.json')
-    ]);
-    const services = fs.readdirSync(outputDir);
+    const { inputDir: inputDirFromConfig, outputDir: outputDirFromConfig } =
+      createOptionsFromConfig(pathToConfig) as {
+        inputDir: string;
+        outputDir: string;
+      };
+    mock({
+      [inputDirFromConfig]: mock.load(inputDirFromConfig),
+      [outputDirFromConfig]: mock.load(outputDirFromConfig),
+      [generatorCommon]: mock.load(generatorCommon),
+      [pathToConfig]: mock.load(pathToConfig),
+      [rootNodeModules]: mock.load(rootNodeModules)
+    });
+    await generate(
+      createOptions({
+        inputDir: inputDirFromConfig,
+        outputDir: outputDirFromConfig,
+        generateJs: true,
+        generatePackageJson: true
+      })
+    );
+    const services = fs.readdirSync(outputDirFromConfig);
     expect(services.length).toBeGreaterThan(0);
-    const entities = fs.readdirSync(path.resolve(outputDir, services[0]));
+    const entities = fs.readdirSync(
+      path.resolve(outputDirFromConfig, services[0])
+    );
     expect(entities).toContain('TestEntity.ts');
+    expect(entities).toContain('TestEntity.js');
     expect(entities).toContain('package.json');
-  }, 60000);
-
+  });
   it('should set version when versionInPackageJson option is used', async () => {
-    const process = await execa('npx', [
-      'ts-node',
-      pathToGenerator,
-      '-c',
-      path.resolve(__dirname, 'generator.config.json'),
-      '--versionInPackageJson=42.23'
-    ]);
-
+    const { inputDir: inputDirFromConfig, outputDir: outputDirFromConfig } =
+      createOptionsFromConfig(pathToConfig) as {
+        inputDir: string;
+        outputDir: string;
+      };
+    mock({
+      [inputDirFromConfig]: mock.load(inputDirFromConfig),
+      [outputDirFromConfig]: mock.load(outputDirFromConfig),
+      [generatorCommon]: mock.load(generatorCommon),
+      [pathToConfig]: mock.load(pathToConfig),
+      [rootNodeModules]: mock.load(rootNodeModules)
+    });
+    await generate(
+      createOptions({
+        inputDir: inputDirFromConfig,
+        outputDir: outputDirFromConfig,
+        generateJs: true,
+        generatePackageJson: true,
+        versionInPackageJson: '42.23'
+      })
+    );
+    const services = fs.readdirSync(outputDirFromConfig);
     const actualPackageJson = JSON.parse(
-      fs.readFileSync(`${outputDir}/test-service/package.json`).toString()
+      fs
+        .readFileSync(
+          path.resolve(
+            outputDirFromConfig.toString(),
+            services[0],
+            'package.json'
+          )
+        )
+        .toString()
     );
     expect(actualPackageJson.version).toEqual('42.23');
-
-    expect(process.stdout).toMatch(
-      /The option 'versionInPackageJson' is deprecated since v2.6.0./
-    );
-  }, 60000);
+  });
+  it('should throw a warning message for a deprecated option even when the generation process is failed', async () => {
```

Comment:
[req] This test times out consistently for me. Maybe we can increase the timeout here ?